### PR TITLE
fix: incorrect host alias symlink on some platforms

### PR DESCRIPTION
**Pull request by kormide**
Fixes https://github.com/abrisco/rules_helm/issues/78.

### CODE REVIEW

File: helm/repositories.bzl [hunk 1]
```diff
@@ -124,12 +125,9 @@ exports_files(["helm{ext}"])
         ext = ext,
     ))
 
-    platform = "{}_{}".format(repository_ctx.os.name, repository_ctx.os.arch)
-    platform_repo_name = _helm_bin_repo_name(platform)
```
Comment (kormide) [inline helm/repositories.bzl:128]: One of the bugs here (and I must have not been testing under bzlmod) is that we need to use the actual name from the repository context to properly capture the bzlmod repo path in the symlink.

File: helm/repositories.bzl [hunk 2]
```diff
@@ -124,12 +125,9 @@ exports_files(["helm{ext}"])
...
-
-    repository_ctx.symlink("../{platform_repo_name}/helm{ext}".format(
+    repository_ctx.symlink("../{name}_{platform}/helm{ext}".format(
         name = repository_ctx.attr.name,
-        platform_repo_name = platform_repo_name,
+        platform = repo_utils.platform(repository_ctx),
```
Comment (kormide) [inline helm/repositories.bzl:130]: This is a nice util from `aspect_bazel_lib`. I can vendor the implementation if you don't want to add an explicit dependency on it. I realize there is a dependency currently, but that's due to the rules_oci bug.

Comment (abrisco) [reply helm/repositories.bzl:130]: I would much prefer to keep the set of dependencies minimal if possible 🙏

**Review by abrisco — CHANGES_REQUESTED** [review]

File: helm/repositories.bzl (hunk 2 above)
Comment (kormide) [reply helm/repositories.bzl:130]: https://github.com/abrisco/rules_helm/pull/79#discussion_r1597540398

Comment (kormide) [reply helm/repositories.bzl:130]: Vendored under 3rdparty.

**Review by abrisco — APPROVED** [review]
Thank you!

**Pull request merged**